### PR TITLE
Frontend-user roles securing

### PR DESCRIPTION
**Pull request by SeriousHorncat**
## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] My code follows the style guidelines enforced by static analysis tools.
- [x] If it is a core feature, I have added thorough tests.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.

## Pull Request Details

Wrike Ticket - [Frontend-user roles securing](https://www.wrike.com/open.htm?id=1175899238)

Changes made:

- Updates to components to rely on user scopes for read/write permissions to allow updates
- API endpoints secured to rely on a user's scope to allow read/write operations

**To Review:**

- [x] Static Analysis by Reviewer
- [x] Deploy local rosalution, with the 'vrr-prep' user, verify images can get attached to section images; login as 'researcher' and verify section images cannot get attached/added

- [x] Verify migration script for existing types of user scopes transition to 'write' permission and a user with 'read' permissions.
  To check this run the following commands:

  1. Download the archived production database to your local machine.  See @SeriousHorncat for the URL to download it from Box.

  2. Import the production database backup to Rosalution from the rosalution root directory

  ```bash
  # ./etc/database/restore-database.sh <container name> <absolute-file-path-to-backup-archive>
  ./etc/database/restore-database.sh rosalution-rosalution-db-1 rosalution-db-2023-07-07-1688746774.archive
  ```
  3. Update User Roles from restored production database

  ```bash
  docker exec rosalution-rosalution-db-1 mongosh /tmp/fixtures/migrate-to-streamlined-user-scopes.js
  ```

- [x] List any other checks needed to be reviewed.
- [x] All Github Actions checks have passed.

### CODE REVIEW

**Comment by JmScherer** [comment]
Attach icons missing on the researcher user, but was able to add via vrr-prep

Screenshot 2023-08-14 at 12 51 50 PM

Screenshot 2023-08-14 at 12 50 32 PM

**Comment by JmScherer** [comment]
The actual command, the PR missed the '/migrations/' portion of the command: `docker exec rosalution-rosalution-db-1 mongosh tmp/fixtures/migrations/migrate-to-streamlined-user-scopes.js`

Screenshot 2023-08-14 at 1 08 43 PM

**Review by fatimarabab — APPROVED** [review]
Works as intended. Good to merge!

**Review by JmScherer — APPROVED** [review]
Looks good!

File: backend/src/security/security.py [hunk 1]
```diff
@@ -128,19 +128,22 @@ def get_authorization(
...
         payload = jwt.decode(token, settings.rosalution_key, algorithms=[settings.oauth2_algorithm])
-        username: str = payload.get("sub")
-        if username is None:
+        client_id: str = payload.get("sub")
+        if client_id is None:
             raise credentials_exception
-        token_scopes = payload.get("scopes", [])
-        token_data = TokenData(scopes=token_scopes, username=username)
+        user_scopes = payload.get("scopes", str)
+        print(user_scopes)
```
Comment (JmScherer) [inline backend/src/security/security.py:135]:
```suggestion
```

**Review by JmScherer — COMMENTED** [review]
Left overs

**Review by JmScherer — APPROVED** [review]

**Pull request merged**